### PR TITLE
DBZ-7138 Documentation for additional-conditions for MongoDb incremental snapshot

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-7138

### CODE REVIEW

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc [hunk 1]
```diff
@@ -83,6 +84,13 @@ The array lists regular expressions which match {data-collection}s by their full
 |An optional `type` component of the `data` field of a signal that specifies the type of snapshot operation to run. +
 Currently supports the `incremental` and `blocking` types. +
 If you do not specify a value, the connector runs an incremental snapshot.
+
+|6
+|`additional-conditions`
+| An optional array that specifies a set of additional conditions that the connector evaluates to determine the subset of records to include in a snapshot. +
+Each additional condition is an object with `data-collection` and `filter` properties.
+You can specify different filters for each data collection. +
+* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:93]: It seems that something is missing here. The asterisk (`*`) is supposed to represent a bullet in an unordered list.  I think that this text is supposed to mirror the text in the description for `additional-conditions` that's in the file `proc-triggering-an-incremental-snapshot-kafka.adoc` (L36-44). Is that right?
That description includes a definition list with two entries, `data-collection` and `filter`. 

To modify the text in this PR to reflect that definition, it's necessary to add the `a` style operator to the specifier for the description column in the table's attributes list on L56 (or place the operator just in front of the cell separator on L90  (i.e., `a|`). 
With that change in place, we can then replace the current text in the table cell with a definition list, by adding the following:

```suggestion
For each additional condition object you can set the following parameters:

data-collection::
The fully-qualified name of the collection that the filter applies to. 
You can apply different filters to each collection. 

filter:: Specifies column values that must be present in a database record for the snapshot to include it, for example, `"color='blue'"`.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
Hi @ani-sha 
A few suggestions and even more questions!

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (ani-sha) [reply documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:93]: There is no such `snapshot. collection.statement.overrides` property. You are right the filter parameter accepts value similar to `snapshot.select.statement.overrides`. I will modify and add it here too.

Comment (ani-sha) [reply documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:93]: > I think that this text is supposed to mirror the text in the description for additional-conditions that's in the file proc-triggering-an-incremental-snapshot-kafka.adoc (L36-44). Is that right?

I think it shouldn't match the kafka adoc. It should rather match the `additional-conditions` of `proc-triggering-an-incremental-snapshot.adoc` (L127-132).

[the remark this answers is not in the document]
Comment (ani-sha) [reply documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:93]: Update : so the details on the `filter` are mentioned on the `proc-triggering-an-incremental-snapshot-kafka.adoc`. And this part of the documentation just points to it. So we don't need to re-mention it again here.

**Comment by ani-sha** [comment]
@roldanbob I have addressed your questions and adjusted the docs. Please let me know if you have any other suggestions.

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc [hunk 2]
```diff
@@ -83,6 +84,13 @@ The array lists regular expressions which match {data-collection}s by their full
 |An optional `type` component of the `data` field of a signal that specifies the type of snapshot operation to run. +
 Currently supports the `incremental` and `blocking` types. +
 If you do not specify a value, the connector runs an incremental snapshot.
+
+|6
+|`additional-conditions`
+| An optional array that specifies a set of additional conditions that the connector evaluates to determine the subset of records to include in a snapshot. +
+Each additional condition is an object with `data-collection` and `filter` properties. You can specify different filters for each data collection. +
+* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:92]:
```suggestion
Each additional condition is an object with `data-collection` and `filter` properties. You can specify different filters for each data collection. +
* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
```
I think that we still have a problem in this passage. L91 states that _"Each additional condition is an object with `data-collection` and `filter` properties"_.  I think that sentence is intended to introduce the unordered list that follows, but the list contains only the single entry, `data-collection`, and no entry for the `filter` property. 

If it is correct that we don't also have a `filter` property, the reference to it should be removed on L91. Or if you set the filter elsewhere, provide a reference to where and how that's done. 

If there's no `filter` property L91 should omit the mention it, and we should remove the asterisk/bullet on L92, letting it just stand as a plain sentence. Currently, we signal an unordered list, and provide an introduction that suggests the list will include two entries, but then we only provide the first item. 
Does that make sense? Let me know if you want to discuss further.

**Review by roldanbob — CHANGES_REQUESTED** [review]
Thanks for your work, @ani-sha. I added one further comment concerning the content on L91-92.

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc (hunk 2 above)
Comment (roldanbob) [reply documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:92]: `data-collection`:: The fully-qualified name of the data collection for which the filter will be applied. 
`filter`:: Specifies the column values that must be present in a data collection record for the snapshot to include it, for example, `"color='blue'"`.

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc [hunk 3]
```diff
@@ -88,9 +88,8 @@ If you do not specify a value, the connector runs an incremental snapshot.
 |6
 |`additional-conditions`
 | An optional array that specifies a set of additional conditions that the connector evaluates to determine the subset of records to include in a snapshot. +
-Each additional condition is an object with `data-collection` and `filter` properties. You can specify different filters for each data collection. +
-* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
-For more information about the `additional-conditions` parameter, see xref:{context}-incremental-snapshots-additional-conditions[].
+`data-collection`:: The fully-qualified name of the data collection for which the filter will be applied.
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-nosql.adoc:91]: Restore sentence that introduces the array keys.

```suggestion
Each element in the `additional-conditions` array is an object that includes the following keys:

`data-collection`:: The fully-qualified name of the data collection for which the filter will be applied.
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc [hunk 4]
```diff
@@ -127,9 +127,8 @@ If you do not specify a value, the connector runs an incremental snapshot.
 |6
 |`additional-conditions`
 | An optional array that specifies a set of additional conditions that the connector evaluates to determine the subset of records to include in a snapshot. +
-Each additional condition is an object with `data-collection` and `filter` properties. You can specify different filters for each data collection. +
-* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
-For more information about the `additional-conditions` parameter, see xref:{context}-incremental-snapshots-additional-conditions[].
+`data-collection`:: The fully-qualified name of the data collection for which the filter will be applied.
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc:130]:
```suggestion
Each element in the `additional-conditions` array is an object that includes the following keys:

`data-collection`:: The fully-qualified name of the data collection for which the filter will be applied.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
@ani-sha One final suggestion. Thanks!

**Review by roldanbob — APPROVED** [review]
LTGM! Thanks!

**Pull request merged**